library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -323,19 +325,24 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
         continue;
       }
 
+      // If we've reached this point then there is data to process; flag that work has been done.
+      captionDataProcessed = true;
+
       boolean previousCaptionValid = captionValid;
       captionValid = (ccHeader & CC_VALID_FLAG) == CC_VALID_FLAG;
       if (!captionValid) {
         if (previousCaptionValid) {
           // The encoder has flipped the validity bit to indicate captions are being turned off.
           resetCueBuilders();
-          captionDataProcessed = true;
         }
         continue;
       }
 
-      // If we've reached this point then there is data to process; flag that work has been done.
-      captionDataProcessed = true;
```

Comment:
I moved this line above, as I think it should be set for any invalid caption as well. That will case the screen update, I do not see a reason why we do not want to update the screen for invalid captions.

Reply:
What does doing this fix? `captionDataProcessed` is already set to `true` on L332 if there's a `valid->invalid` transition. For `invalid->invalid` the screen will already be blank, so updating the screen wont do anything useful?

Reply:
I just wanted to reduce redundancy, instead of setting the flag inside and after this in block, we can achieve the same by setting it before the block, can't we?

Reply:
Updated.